src/main/java/com/mael/payroll/affiliations/UnionAffiliation.java

```diff
@@ -0,0 +1,57 @@
+package com.mael.payroll.affiliations;
+
+import com.mael.payroll.cards.AffiliationCard;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UnionAffiliation implements Affiliation {
+
+    private double fees;
+    private List<AffiliationCard> affiliationCards;
+
+    public UnionAffiliation(double fees) {
+        this.fees             = fees;
+        this.affiliationCards = new ArrayList<>();
+    }
+
+    @Override
+    public double getFees() {
+        return fees;
+    }
+
+    public void addAffiliationCard(AffiliationCard affiliationCard) {
+        this.affiliationCards.add(affiliationCard);
+    }
+
+    public AffiliationCard getAffiliationCard(LocalDate date) {
```

Comment:
Similar to what we discussed today, this method is used only in tests.
I'd rewrite the tests in a way that you get all affiliation cards back and have the assertions on that. Instead of this getter that also introduces an exception that doesn't seem to be needed.